packages/vscode-tailwindcss/src/extension.ts

```diff
@@ -297,147 +290,129 @@ export async function activate(context: ExtensionContext) {
   // not just the language IDs
   // e.g. "plaintext" already exists but you change it from "html" to "css"
   context.subscriptions.push(
-    Workspace.onDidChangeConfiguration((event) => {
-      let toReboot = new Set<WorkspaceFolder>()
-
-      Workspace.textDocuments.forEach((document) => {
-        let folder = Workspace.getWorkspaceFolder(document.uri)
-        if (!folder) return
-        if (event.affectsConfiguration('tailwindCSS.experimental.configFile', folder)) {
-          toReboot.add(folder)
-        }
+    Workspace.onDidChangeConfiguration(async (event) => {
+      let folders = Workspace.workspaceFolders ?? []
+
+      let needsReboot = folders.some((folder) => {
+        return (
+          event.affectsConfiguration('tailwindCSS.experimental.configFile', folder) ||
+
+          // TODO: Only reboot if the MAPPING changed instead of just the languages
+          // e.g. "plaintext" already exists but you change it from "html" to "css"
+          // TODO: This should not cause a reboot of the server but should instead
+          // have the server update its internal state
+          event.affectsConfiguration('tailwindCSS.includeLanguages', folder)
+        )
       })
-      ;[...clients].forEach(([key, client]) => {
-        const folder = Workspace.getWorkspaceFolder(Uri.parse(key))
-        if (!folder) return
-
-        let reboot = false
-
-        if (event.affectsConfiguration('tailwindCSS.includeLanguages', folder)) {
-          const userLanguages = getUserLanguages(folder)
-          if (userLanguages) {
-            const userLanguageIds = Object.keys(userLanguages)
-            const newLanguages = dedupe([...defaultLanguages, ...userLanguageIds])
-            if (!equal(newLanguages, languages.get(folder.uri.toString()) ?? [])) {
-              languages.set(folder.uri.toString(), newLanguages)
-              reboot = true
-            }
-          }
-        }
 
-        if (event.affectsConfiguration('tailwindCSS.experimental.configFile', folder)) {
-          reboot = true
-        }
+      if (!needsReboot) {
+        return
+      }
 
-        if (reboot && client) {
-          toReboot.add(folder)
-        }
-      })
+      // IDEA: We might be able to service requests during a reboot by booting the
```

Comment:
Is this todo/idea still relevant, or do we want to tackle this in a separate PR?

Reply:
It is but it's for something to be done later. I'll drop this and add a card for it.